snuba/consumers/consumer_builder.py

```diff
@@ -83,62 +83,98 @@ def __init__(
         mock_parameters: Optional[MockParameters] = None,
         cooperative_rebalancing: bool = False,
     ) -> None:
-        self.storage = get_writable_storage(storage_key)
+        self.storages = {key: get_writable_storage(key) for key in storage_keys}
         self.bootstrap_servers = kafka_params.bootstrap_servers
         self.consumer_group = kafka_params.group_id
-        topic = (
-            self.storage.get_table_writer()
+
+        # in multistorage case, we ensure there is one topic in common
+        topics = {
+            storage.get_table_writer()
             .get_stream_loader()
             .get_default_topic_spec()
             .topic
-        )
+            for storage in self.storages.values()
+        }
+        topic = topics.pop()
+        if topics:
+            raise ValueError("only one topic is supported")
+
+        if len(self.storages) == 1:
+            self.broker_config = get_default_kafka_configuration(
+                topic, bootstrap_servers=kafka_params.bootstrap_servers
+            )
+            logger.info(
+                f"librdkafka log level: {self.broker_config.get('log_level', 6)}"
+            )
 
-        self.broker_config = get_default_kafka_configuration(
-            topic, bootstrap_servers=kafka_params.bootstrap_servers
-        )
-        logger.info(f"librdkafka log level: {self.broker_config.get('log_level', 6)}")
-        self.producer_broker_config = build_kafka_producer_configuration(
-            topic,
-            bootstrap_servers=kafka_params.bootstrap_servers,
-            override_params={
-                "partitioner": "consistent",
-                "message.max.bytes": 50000000,  # 50MB, default is 1MB
-            },
-        )
+            self.producer_broker_config = build_kafka_producer_configuration(
+                topic,
+                bootstrap_servers=kafka_params.bootstrap_servers,
+                override_params={
+                    "partitioner": "consistent",
+                    "message.max.bytes": 50000000,  # 50MB, default is 1MB
+                },
+            )
 
-        stream_loader = self.storage.get_table_writer().get_stream_loader()
+            # XXX: This can result in a producer being built in cases where it's
+            # not actually required.
+            self.producer = Producer(self.producer_broker_config)
 
-        self.raw_topic: Topic
+        storage_keys = [*self.storages.keys()]
+
+        self.raw_topic: ArroyoTopic
```

Comment:
Why calling this ArroyoTopic?

Reply:
There's a mix of Topic object types in this file. Some are Snuba topics, some are Arroyo topics, and some functions can only take in Snuba Topics and some Arroyo Topics. I figured that it would be helpful to distinguish when we're initializing an ArroyoTopic so that we don't mistake the two/pass one in when we meant the other.